language_detector/main.py

```diff
@@ -1,9 +1,28 @@
-"""This is the entry point of the program."""
-
 from .languages import LANGUAGES
 
-
 def detect_language(text, languages=LANGUAGES):
     """Returns the detected language of given text."""
-    # implement your solution here
-    pass
+    text_lang = "Unknown"
+    a_list = text.split()
+    
+    # For each language, check every word and see if it can be found in the
+    # languages.py file, if so add that word to a list.
+    # Finally, get the length of each of these lists.
+    try:
+        eng = [x for x in a_list for y in languages[2]['common_words'] if x==y]
+        length_english = len(eng)
+    except IndexError:
+        length_english = 0
+    ger = [x for x in a_list for y in languages[1]['common_words'] if x==y]
+    length_german = len(ger)
+    spa = [x for x in a_list for y in languages[0]['common_words'] if x==y]
+    length_spanish = len(spa)
+
```

Comment:
This list comprehension is used 3 times:
```python
[x for x in a_list for y in languages[NUMBER]['common_words'] if x==y]
```
This can be redone using a for loop that iterates through each item.

Also, `x`, `y`, and `x==y` is very unreadable (a_list is on the fence too). The list comprehension can be rewritten as:
```python
[match_target_word for match_target_word in a_list for common_word in languages[NUMBER]['common_words'] if match_target_word == common word]
```